sdk/core/core-sse/src/sse.ts

```diff
@@ -33,8 +25,51 @@ export interface EventMessage {
 
 type PartialSome<T, K extends keyof T> = Omit<T, K> & Partial<Pick<T, K>>;
 
-export function toSSE(chunkIter: AsyncIterable<Uint8Array>): AsyncIterable<EventMessage> {
-  return toMessage(toLine(chunkIter));
+/**
+ * Processes a response stream into a stream of events.
+ * @param chunkIter - A stream of Uint8Array chunks
+ * @returns An async iterable of EventMessage objects
+ */
+export function toSSE(chunkIter: ReadableStream<Uint8Array>): AsyncIterable<EventMessage>;
+/**
+ * Processes a response stream into a stream of events.
+ * @param chunkIter - An async iterable of Uint8Array chunks
+ * @returns An async iterable of EventMessage objects
+ */
+export function toSSE(chunkIter: AsyncIterable<Uint8Array>): AsyncIterable<EventMessage>;
+export function toSSE(
+  chunkIter: AsyncIterable<Uint8Array> | NodeJS.ReadableStream | ReadableStream<Uint8Array>
+): AsyncIterable<EventMessage> {
+  return toMessage(
+    toLine(
+      isReadableStream(chunkIter) && (chunkIter as any)[Symbol.asyncIterator] === undefined
+        ? toAsyncIterable(chunkIter)
+        : (chunkIter as AsyncIterable<Uint8Array>)
```

Comment:
nit: can we pull this logic out into its own internal function for readability?

Thinking something like

```ts
return toMessage(toLine(ensureAsyncIterable(chunkIter)));
```